ext/java/org/msgpack/jruby/Decoder.java

```diff
@@ -118,7 +120,12 @@ private IRubyObject consumeUnsignedLong() {
   private IRubyObject consumeString(int size, Encoding encoding) {
     byte[] bytes = readBytes(size);
     ByteList byteList = new ByteList(bytes, encoding);
-    return runtime.newString(byteList);
+    RubyString string = runtime.newString(byteList);
+    if (this.freeze) {
+      string.setFrozen(true);
```

Comment:
`Ruby#freezeAndDedupString` looks to freeze&dedup the string, not only dedup.
Is there any reason to call `setFrozen(true)` explicitly here?

Reply:
Yes, today it's a bit useless, however in theory if you call it with a frozen string, it could directly intern that string without duplicating it first. That's what MRI 3.0 now does, see: https://github.com/ruby/spec/pull/788

I do hope to change that behavior in JRuby at some point, so I figured we might as well call it with a frozen string directly.